gha/bin/release-promote.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3.10
+from os import environ
+from typing import NoReturn
+from jira.resources import Issue
+from argparse import ArgumentParser
+
+from gha.jira.assignee import Assignee
+from gha.jira.project import JiraProject
+from gha.jira.issuestatus import IssueStatus
+from gha.jira.basejiracommand import BaseJiraCommand
+from gha.github.basegithubcommand import BaseGithubCommand
+
+DEFAULT_ENV = environ.get("ENV") or None
+CURRENT_RELEASE_JQL = """
+    project = "%(project)s"
+    AND type = "GBL-Release"
+    AND status in (%(statuses)s)
+    ORDER BY created ASC
+"""
+PREPROD_STATUSES = (
+    IssueStatus.InTest,
+    IssueStatus.ReadyForRelease,
+)
+PRETEST_STATUSES = (
+    IssueStatus.UploadingToStaging,
+    IssueStatus.ReadyForTest,
+)
+PRODUCTION_STATUSES = (
+    IssueStatus.UploadingToProduction,
+    IssueStatus.ReadyForRelease,
+)
+
+
+class ReleasePromote(BaseGithubCommand, BaseJiraCommand):
+    def handle(self, env: str, deployed: bool) -> NoReturn:
+        search_statuses: tuple
+        promote_to: str
+        assignee: Assignee
+        if env not in ["prod", "stage"]:
+            raise AssertionError(f'Prohibited environment passed: {env}. Allowed only "prod" or "stage"')
+        repo_name = self.repo.name
+        self.log.info(f"Repository name: {repo_name}")
+        jira_project = self.get_jira_project_for_repo(repo_name)
+        self.log.info(f"JIRA project name: {jira_project}")
+        # Decide how to find relevant release issue,
+        # Where to promote and Who to assigned
+        if deployed:
+            if env == 'prod':
+                search_statuses = PRODUCTION_STATUSES
+                assignee = Assignee.Producer[jira_project]
+                if jira_project == JiraProject.MathmagicUnity:
+                    promote_to = IssueStatus.ReadyForPublication
+                else:
+                    promote_to = IssueStatus.Done
+            else:
+                search_statuses = PRETEST_STATUSES
+                assignee = Assignee.QA[jira_project]
+                promote_to = IssueStatus.ReadyForTest
+        else:
+            search_statuses = PREPROD_STATUSES
+            assignee = Assignee.DevOps[None]
+            promote_to = IssueStatus.UploadingToProduction
+
+        # Find the current release ticket or create a new one
+        s = "', '".join(search_statuses)
+        statuses = f"'{s}'"
+        self.log.info(
+            msg=f'-Env: {env}\n-Search: {statuses}\n-Promote to: {promote_to}\n-Assignee: {assignee}')
+        releases = self.jira.search_issues(
```

Comment:
Вот тут можно сразу тип указать, если реально требуется.

```suggestion
        releases list[Issue] = self.jira.search_issues(
```